### DISCUSSION TITLE
Dynamic routes with SSR lead to 404 pages on visit

### DISCUSSION THREAD

**Discussion by hjpotter92** [vercel/next.js#10990] [Help] [answered]
# Bug report

## Describe the bug

I have a nextjs application, which supports pretty url like: <https://demo.oga.fit/balance> or <https://demo.oga.fit/balance/diva-yoga>. It is exported to work behind gitlab pages.

The slugs are loaded from the backend over a request. The [home page][home] lists them. When going from the home page to the slug based URL, the pages are working. However, if I refresh the browser when already on the above mentioned URL, it shows a 404 page instead.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to <https://demo.oga.fit/>
2. Click on `Balance` or `Learn` images shown
3. After the page is rendered, refresh the browser tab. It should display a 404 error on refresh.

## Expected behavior

Instead of the 404 page, I'd assume that the dynamic routes would come into play and render the actual content.

## Additional context

It is a closed source application; I'd not be able to share code snippets.

  [home]: https://demo.oga.fit/

**Comment by timneutkens** [comment]
> It is a closed source application; I'd not be able to share code snippets.

This makes it impossible to look into this. But my initial thoughts are that you're not providing all paths to be exported. You will probably want to adopt
`getStaticProps` and `getStaticPaths`.

https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation
https://nextjs.org/docs/basic-features/data-fetching#getstaticpaths-static-generation

**Comment by hjpotter92** [comment]
@timneutkens The paths are not constant. They might vary between backend deployments.

**Reply by timneutkens** [reply]
Then you can't use `next export`, you'll have to host on [ZEIT Now](https://nextjs.org/docs/deployment) or use `next start`

**Comment by hjpotter92** [comment, marked as answer]
> Then you can't use `next export`, you'll have to host on [ZEIT Now](https://nextjs.org/docs/deployment) or use `next start`

Moved the project to zeit now deployment. Thanks for the pointers.
